### DISCUSSION TITLE
Ubuntu images: dated codename support

### DISCUSSION THREAD

**Discussion by cdloh** [renovatebot/renovate#25042] [Suggest an Idea]
### Tell us more.

Creating a discussion as per #25004 

#24717 doesn't do everything this does as it still can't do bumps when new distro versions are released where as this will handle it.

**Comment by rarkins** [comment]
First of all, thanks for your PR. Hopefully we can merge it in the same or similar form once satisfied with requirements.

My main confusion here is that both the title of this discussion and your PR imply that only tags with a format like `jammy-yymmdd` are "container images" - why is that? Because from my perspective, ubuntu container images are the full list here https://hub.docker.com/_/ubuntu/tags

Therefore to use the term like "isContainerImage()" in functions where it's technically wrong/overly opinionated results in confusion for future maintainers. Wouldn't a term like `isDatedCodeName()` be more appropriate?

**Reply by cdloh** [reply]
That sounds like a better function name.

Happy to take any further feedback.

Is there a reason why we're doing this in a discussion vs feedback on the MR?

**Comment by rarkins** [comment]
> Is there a reason why we're doing this in a discussion vs feedback on the MR?

https://github.com/renovatebot/renovate/blob/0c5df8ec4600000b068534bc16c9b2c497595186/.github/contributing.md?plain=1#L30-L31

We don't say you MUST because that would be too strict. But in this case, it was necessary.

FYI I have renamed this discussion now so that it's more accurate. Please feel free to reopen or restart the PR reflecting the agreed terminology/topic
